demo-python/code/azure-search-vector-image-index-creation-python-sample.ipynb

```diff
@@ -0,0 +1,389 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install azure-search-documents==11.4.0b11 --quiet\n",
+    "! pip install python-dotenv ipython --quiet\n",
+    "! pip install azure-storage-blob --quiet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import required libraries and environment variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import required libraries  \n",
+    "import os  \n",
+    "import json  \n",
+    "import requests\n",
+    "import http.client, urllib.parse\n",
+    "import shutil\n",
+    "from dotenv import load_dotenv  \n",
+    "from azure.core.credentials import AzureKeyCredential  \n",
+    "from azure.search.documents.indexes import SearchIndexClient, SearchIndexerClient  \n",
+    "\n",
+    "from azure.search.documents.indexes.models import (  \n",
+    " \n",
+    "    ExhaustiveKnnParameters,  \n",
+    "    ExhaustiveKnnVectorSearchAlgorithmConfiguration,\n",
+    "    HnswParameters,  \n",
+    "    HnswVectorSearchAlgorithmConfiguration,\n",
+    "    IndexingParameters,\n",
+    "    SimpleField,\n",
+    "    SearchField,  \n",
+    "    SearchFieldDataType,  \n",
+    "    SearchIndex,  \n",
+    "    SearchIndexer,\n",
+    "    SearchIndexerDataSourceConnection, \n",
+    "    SearchIndexerDataContainer,  \n",
+    "    VectorSearch,  \n",
+    "    VectorSearchAlgorithmKind,  \n",
+    "    VectorSearchProfile,  \n",
+    ")\n",
+    "\n",
+    "from azure.storage.blob import BlobServiceClient\n",
+    "  \n",
+    "load_dotenv()  \n",
+    "service_endpoint = os.getenv(\"AZURE_SEARCH_SERVICE_ENDPOINT\")  \n",
+    "index_name = os.getenv(\"AZURE_SEARCH_INDEX_NAME\")\n",
+    "key = os.getenv(\"AZURE_SEARCH_ADMIN_KEY\")  \n",
+    "aiVisionApiKey = os.getenv(\"AZURE_AI_VISION_API_KEY\")  \n",
+    "aiVisionRegion = os.getenv(\"AZURE_AI_VISION_REGION\")\n",
+    "blob_connection_string = os.getenv(\"BLOB_CONNECTION_STRING\")  \n",
+    "container_name = os.getenv(\"BLOB_CONTAINER_NAME\")\n",
+    "credential = AzureKeyCredential(key)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Image Embedding\n",
+    "\n",
+    "Vectorize images with image embedding API "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_image_vector(image_path, key, region):\n",
+    "    headers = {\n",
+    "        'Ocp-Apim-Subscription-Key': key,\n",
+    "    }\n",
+    "\n",
+    "    params = urllib.parse.urlencode({\n",
+    "        'model-version': '2023-04-15',\n",
+    "    })\n",
+    "\n",
+    "    MAX_RETRIES = 5\n",
```

Comment:
tenacity has a good retry library you can utilize here

https://tenacity.readthedocs.io/en/latest/